src/util/index.ts

```diff
@@ -278,21 +278,12 @@ export const cleanNumberVal = (value: number | string | null): number | null =>
       ? parseFloat(value.trim().replaceAll(',', ''))
       : null;
 
-export const toCamelCase = (originalString: string) => {
-  let isNextLetterUppercase = false;
-  let convertedString = '';
-
-  for (const char of originalString) {
-    if (char === ' ') {
-      isNextLetterUppercase = true;
-      continue;
-    }
-
-    convertedString += isNextLetterUppercase
-      ? char.toUpperCase()
-      : char.toLowerCase();
-    isNextLetterUppercase = false;
-  }
-
-  return convertedString;
-};
+export const toCamelCase = (originalString: string) =>
+  originalString
+    .split(' ')
+    .map((word, index) =>
+      index === 0
+        ? word.toLowerCase()
+        : word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+    )
+    .join('');
```

Comment:
Without `.trim` its test case failing.
 
Replacing multiple space by single space could be save some.
```js
'multiple      spaces'.split(' ') // [ 'multiple', '', '', '', '', '', 'spaces' ]
```

```suggestion
export const toCamelCase = (originalString: string) =>
  originalString
    .trim()
    .replace(/\s+/g, ' ')
    .split(' ')
    .map((word, index) =>
      index === 0
        ? word.toLowerCase()
        : word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
    )
    .join('');
```

Reply:
I was unable to run unit tests on my machine when I was opening the PR, since all test cases were failing, so when I opened a PR and saw that it passed, I though it's all good. Thanks for catching the issue. I will answer to both inline and code review comment here.

For failing unit tests, I'd do the same thing we're doing in hpc_service, where we capture `jest` exit code and then exit the script with that code. I think it is a responsibility of the script to do this correctly, rather than CI to account for shell script's misbehavior.

As for the `toCamelCase` method, we indeed need `trim()`, because leading spaces would be treated as first character with the current algorithm. On the other hand, I don't know whether replacing multiple spaces by single space saves time, because the call to `replace` is not free, although I don't know what is its time and space complexity. But, if it is O(n), maybe we don't save much.

I have made optimizations to `toCamelCase` method by measuring execution times, which I have done again. I needed to revert the "caching" which actually brings the significant boost in performance. Turns out that using `replace` adds 7-9 seconds to execution time. I have measured without any changes to my proposed `toCamelCase` method, then with `trim()` added and with `replace` added after that. Each of these were measured three times. Trimming doesn't add anything significantly, but `replace` impacts performance.

Now, as I said, this will not be felt because of "caching", but since method `toCamelCase` is being optimized in one of the commits, I will not add `replace`, since it hurts performance.